Replicator/c4Replicator.cc

```diff
@@ -62,9 +62,15 @@ static uint16_t defaultPortForScheme(slice scheme) {
 bool c4repl_isValidDatabaseName(C4String dbName) C4API {
     slice name = dbName;
     // Same rules as Couchbase Lite 1.x and CouchDB
-    return name.size > 0 && name.size < 240
+    const auto valid = name.size > 0 && name.size < 240
         && islower(name[0])
         && !slice(name).findByteNotIn("abcdefghijklmnopqrstuvwxyz0123456789_$()+-/"_sl);
+
+    if(!valid) {
+        WarnError("%.*s is not a valid database name for replication", SPLAT(dbName));
```

Comment:
This seems the wrong place to log this. It's simply an informational function. There must be a function that calls this and, on false, causes an error; that would be the place to log.